Add Google Authentication for experimental API (#9848)
mistercrunch/Airflow: docs/security.rst

The Kerberos service is configured as ``airflow/fully.qualified.domainname@REALM``. Make sure this
principal exists in the keytab file.

You can also configure
`Google OpenID <https://developers.google.com/identity/protocols/oauth2/openid-connect>`__
for authorization. To enable it, set the following option in the configuration:

.. code-block:: ini

    [api]
    auth_backend = airflow.providers.google.common.auth_backend.google_openid

> It is also highly recommended to configure an OAuth2 audience so that the generated used tokens can only

Reviewer: ```suggestion
It is also highly recommended to configure an OAuth2 audience so that the generated tokens can only
```

Reviewer 2: ```suggestion
It is also highly recommended to configure an OAuth2 audience so that the generated tokens are restricted to use by Airflow only.
```
